lib/wallet/http.js

```diff
@@ -1153,6 +1502,77 @@ class HTTPOptions {
   }
 }
 
+class TransactionOptions {
+  /**
+   * TransactionOptions
+   * @alias module:http.TransactionOptions
+   * @constructor
+   * @param {Validator} valid
+   */
+
+  constructor(valid) {
+    if (valid)
+      return this.fromValidator(valid);
+  }
+
+  /**
+   * Inject properties from Validator.
+   * @private
+   * @param {Validator} valid
+   * @returns {TransactionOptions}
+   */
+
+  fromValidator(valid) {
+    assert(valid);
+
+    this.rate = valid.u64('rate');
+    this.maxFee = valid.u64('maxFee');
+    this.selection = valid.str('selection');
+    this.smart = valid.bool('smart');
+    this.account = valid.str('account');
+    this.locktime = valid.u64('locktime');
+    this.sort = valid.bool('sort');
+    this.subtractFee = valid.bool('subtractFee');
+    this.subtractIndex = valid.i32('subtractIndex');
+    this.depth = valid.u32(['confirmations', 'depth']);
+    this.outputs = [];
+
+    if (valid.has('outputs'))
+      this.setOutputs(valid);
+
+    return this;
+  }
+
+  setOutputs(valid) {
```

Comment:
Is there a reason we moved this logic into a method? If we do not plan to call this elsewhere, it makes sense to live this inside of `fromValidator`. 

If we leave this as a separate method, we should validate the input, i.e., `valid instanceof Validator`.

Reply:
I separated the 2 because I thought it looked better aesthetically. Moved it back in though, not very opinionated on it.